Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -2470,16 +2470,16 @@ NonnullRefPtr<FunctionNodeType> Parser::parse_function_node(u8 parse_options)
         function_kind = FunctionKind::Normal;
     String name;
     if (parse_options & FunctionNodeParseOptions::CheckForFunctionAndName) {
-        if (function_kind == FunctionKind::Normal && match(TokenType::Async) && !next_token().trivia_contains_line_terminator()) {
```

Comment:
Is this change intended?

Reply:
Yeah this might mess with the valid
`async function async() { await 1; }`
Which we don't have a test for... Whoops that's my bad

Reply:
Sure, but it doesn't seem related to the commit which changes it.

Reply:
Yeah that as well

Reply:
ah, that's leftover from debugging. Initially I was passing parse_options, but turns out it doesn't consume the async token anymore if it already determined `function_kind = FunctionKind::Async` from those options :acid2:
Reverted :+1: